apps/backend/src/event/entities/event.entity.ts

```diff
@@ -0,0 +1,64 @@
+import { Category, Group, Priority, Status, User } from '@prisma/client';
+import { IsDate, IsDateString, IsEnum, IsInt, IsNotEmpty, IsOptional, IsString } from 'class-validator';
+
+export class Event {
+  @IsInt()
+  @IsOptional()
+  id: number;
+
+  @IsString()
+  @IsNotEmpty()
+  name: string;
+
+  @IsString()
+  @IsOptional()
+  description: string;
+
+  @IsString()
+  @IsOptional()
+  location: string;
+
+  @IsDate()
+  startDate: Date;
+
+  @IsDate()
+  @IsOptional()
+  endDate: Date;
+
+  @IsDateString()
+  @IsOptional()
+  startTime: Date;
+
+  @IsDateString()
+  @IsOptional()
+  endTime: Date;
+
+  @IsEnum(Priority)
+  @IsOptional()
+  priority: Priority;
+
+  @IsEnum(Priority)
+  @IsOptional()
+  status: Status;
+
+  @IsInt()
+  @IsNotEmpty()
+  categoryId: number;
+
+  @IsNotEmpty()
+  category: Category;
+
+  @IsInt()
+  @IsOptional()
+  ownerUserId: number;
+
+  @IsOptional()
+  ownerUser: User;
+
+  @IsInt()
+  @IsOptional()
+  ownerGroupId: number;
+
+  @IsOptional()
+  ownerGroup: Group;
```

Comment:
Do not create a field for relations in the entity schema. This class should represent one record in the database. Prisma hides this from you, but there are no ownerGroup, ownerUser or category fields on the Event table (only categoryId, ownerUserId and ownerGroupId), so you should remove those from this class. When an event is created, the user, group and category likely already exists, and prisma will be able to link the new event record to those records with the id (foreign key) fields